lib/blocks.php

```diff
@@ -172,6 +172,40 @@ function gutenberg_reregister_core_block_types() {
 
 add_action( 'init', 'gutenberg_reregister_core_block_types' );
 
+/**
+ * Adds the defer loading strategy to all registered blocks.
+ *
+ * This function would not be part of core merge. Instead, the register_block_script_handle() function would be patched
+ * as follows.
+ *
+ * ```
+ * --- a/wp-includes/blocks.php
+ * +++ b/wp-includes/blocks.php
+ * @ @ -153,7 +153,8 @ @ function register_block_script_handle( $metadata, $field_name, $index = 0 ) {
+ *                 $script_handle,
+ *                 $script_uri,
+ *                 $script_dependencies,
+ * -           isset( $script_asset['version'] ) ? $script_asset['version'] : false
+ * +         isset( $script_asset['version'] ) ? $script_asset['version'] : false,
+ * +         array( 'strategy' => 'defer' )
+ *         );
+ *         if ( ! $result ) {
+ *                 return false;
+ * ```
+ *
+ * @see register_block_script_handle()
+ */
+function gutenberg_defer_block_view_scripts() {
+	$block_types = WP_Block_Type_Registry::get_instance()->get_all_registered();
+	foreach ( $block_types as $block_type ) {
+		foreach ( $block_type->view_script_handles as $view_script_handle ) {
+			wp_script_add_data( $view_script_handle, 'strategy', 'defer' );
+		}
+	}
```

Comment:
Shouldn't this be enforced only for Core blocks? We have no idea what 3rd-party blocks may be doing...
We could add a condition to only process Core blocks:

```suggestion
	foreach ( $block_types as $block_type ) {
		if ( ! str_starts_with( $block_type->name, 'core/' ) ) {
			continue;
		}
		foreach ( $block_type->view_script_handles as $view_script_handle ) {
			wp_script_add_data( $view_script_handle, 'strategy', 'defer' );
		}
	}
```

Reply:
What is the drawback of enabling for non-core blocks? I agree with your general sentiment, but we have considered the implications of this change, and there should be no breakage: Given that these JS files were already loaded in either head or footer (head for block themes, footer for classic themes), any block using them already needed to write the JS code in a way that works regardless of where the file is loaded. So deferring is safe here and should only bring benefits.

Reply:
Agreed

Reply:
Thanks for the additional context! 
I always try to be a bit cautious which is why I suggested adding a condition, it just felt safer that way.
If we are certain that there's no way a plugin developer has messed this up on their own blocks (we can't always assume they do things reasonably or well), I'll go ahead and mark this as resolved 👍